src/bot/logic/common/rulesCache.ts

```diff
@@ -0,0 +1,46 @@
+import { GameApi, TechnoRules } from "@chronodivide/game-api";
+
+// checking technorules directly reduces the amount of calls to getUnitData(), which is a relatively expensive function.
+// A null value indicates an object that does not have TechnoRules.
+const technoRulesCache: { [rulesName: string]: TechnoRules | null } = {};
```

Comment:
Rules can change between game instances, but the cache is static and cache entries are never evicted.